NGS2_WITNESS_Cycle1_difi_prep_exp2.R

```diff
@@ -0,0 +1,218 @@
+  return(breadboard.frame)
+}
+
+# Extract DIFI scores
+
+DIFI.final<-breadboard.extract ("DIFI")
+
+# Compute individual probability of cooperation
+
+av.cooperation.exp2 <-aggregate(exp2_cooperation$decision0d1c,
+                                by=list(exp2_cooperation$playerid),
+                                FUN=mean,
+                                na.rm=TRUE)
+colnames(av.cooperation.exp2)<-c("pid", "prob.cooperate")
+
+### Create frame of nodes in network, with their final score and final group.
+
+# Final score
+
+score.exp2 <- breadboard.extract ("FinalScore")
+
+# Final group
+
+exp2_cooperation <- read.csv('NGS2-Cycle1-Experiment2/cooperation_exp2.csv',
+                             header = TRUE, sep = ',')
+
+final.group <- unique(subset (exp2_cooperation, 
+                       round_num == 15, 
+                       select = c('playerid', 'group')))
+
+colnames(final.group) <- c('pid', 'group')
+
+### Add network properties (cooperative structure)
+
+connections.list <- lapply(exp2, function(x) {
+  roll <- subset(x, 
+                 event == "Connected" | event == "Disconnected" , 
+                 select = c('id',
+                            "event",
+                            'data.name', 
+                            'data.value')
+  )
+  return(roll)
+})
+
+connections.list2 <- mapply(`[<-`, 
+                            connections.list, 
+                           'session', 
+                           value = names(breadboard.list), 
+                           SIMPLIFY = FALSE)
+connection.events<-do.call(rbind,
+                           lapply(connections.list2,data.frame)
+)
+connections.exp2.dcast <- dcast(connection.events,
+                                id+session+event ~ data.name,
+                                value.var="data.value")
+
+connections.exp2.final <- connections.exp2.dcast[0,]
+
+for (i in 1:length(connections.exp2.dcast$id))
+{
+  if
+  (connections.exp2.dcast$event[i]=='Connected'){
+    connections.exp2.final[i,] <- rbind(connections.exp2.dcast[i,])
+  }
+  else if
+  (connections.exp2.dcast$event[i]=='Disconnected'){
+    connections.exp2.final <-
+      subset(connections.exp2.final, !(
+        (playerId1 == connections.exp2.dcast[i, c("playerId1")] &
+           playerId2 == connections.exp2.dcast[i, c("playerId2")]) |
+          (playerId1 == connections.exp2.dcast[i, c("playerId2")] &
+             playerId2 == connections.exp2.dcast[i, c("playerId1")])
+      )
+      )
+  }
+  else {
+    NULL
+  }
+}
+
+connections.from.to.exp2 <- cbind(connections.exp2.final$playerId1,
+                                  connections.exp2.final$playerId2, 
+                                  connections.exp2.final$session)
+
+
+connections.from.to.exp2.final <- connections.from.to.exp2[complete.cases(connections.from.to.exp2), ]
+DIFI.1 <- merge (av.cooperation.exp2, score.exp2, by = "pid")
+
+DIFI.1.subset<-subset(DIFI.1, session == i)

```

Comment:
This is strange -- `i` is going to be whatever the last iteration value from the `for()` loop above. I don't think that's what you want.

Reply:
Actually, seeing what's going on below, I think this is errant code and should be deleted.

Reply:
It should, thanks!